client/src/components/UI_LoadingWrapper.re

```diff
@@ -8,39 +8,44 @@ type action =
   | ChangeLoading(bool)
   | ChangeLoadingDisplay(bool);
 
-let component = ReasonReact.reducerComponent("UI_LoadingWrapper");
+let reducer = (state, action) =>
+  switch (action) {
+  | ChangeLoadingDisplay(loading) => {...state, displayLoading: loading}
+  | ChangeLoading(loading) => {...state, loading}
+  };
 
 let clearTimeout =
   fun
   | None => ()
   | Some(timeoutId) => Js.Global.clearTimeout(timeoutId);
 
-let make = (~delayMs=1000, ~loading: bool, children) => {
-  ...component,
-  initialState: () => {
-    loading,
-    displayLoading: loading,
-    timeoutId: ref(None),
-  },
-  willUnmount: self => clearTimeout(self.state.timeoutId^),
-  willReceiveProps: self =>
-    if (loading != self.state.loading) {
-      {...self.state, loading};
-    } else {
-      self.state;
-    },
-  didUpdate: ({oldSelf, newSelf: self}) =>
-    if (oldSelf.state.loading != loading) {
+[@react.component]
+let make = (~delayMs=1000, ~loading: bool, ~children) => {
+  let (state, dispatch) =
+    React.useReducer(
+      reducer,
+      {loading, displayLoading: loading, timeoutId: ref(None)},
+    );
+
+  let {loading, displayLoading, timeoutId} = state;
+
+  React.useEffect(() => {
+    clearTimeout(timeoutId^);
```

Comment:
If you look carefully at the previous code, this call to `clearTimeout` was done when unmounting. To keep the same behavior, it should be moved to the returned value of the effect, e.g.

```
 React.useEffect(() => {
    Some(() => clearTimeout(timeoutId^));
}
```

Reply:
Thanks! Changed to return some value.